hooks/tk-multi-publish2/basic/collector.py

```diff
@@ -0,0 +1,330 @@
+                    }
+                }
+            )
+
+            self.collect_playblasts(item, project_root)
+            self.collect_alembic_caches(item, project_root)
+
+        else:
+
+            self.logger.info(
+                "Could not determine the current Maya project.",
+                extra={
+                    "action_button": {
+                        "label": "Set Project",
+                        "tooltip": "Set the Maya project",
+                        "callback": lambda: mel.eval('setProject ""')
+                    }
+                }
+            )
+
+    def collect_current_maya_session(self, settings, parent_item):
+        """
+        Creates an item that represents the current maya session.
+
+        :param parent_item: Parent Item instance
+
+        :returns: Item of type maya.session
+        """
+
+        publisher = self.parent
+
+        # get the path to the current file
+        path = cmds.file(query=True, sn=True)
+
+        # determine the display name for the item
+        if path:
+            file_info = publisher.util.get_file_path_components(path)
+            display_name = file_info["filename"]
+        else:
+            display_name = "Current Maya Session"
+
+        # create the session item for the publish hierarchy
+        session_item = parent_item.create_item(
+            "maya.session",
+            "Maya Session",
+            display_name
+        )
+
+        # get the icon path to display for this item
+        icon_path = os.path.join(
+            self.parent.engine.disk_location,
+            "resources",
+            "maya.png"
+        )
+        session_item.set_icon_from_path(icon_path)
+
+        # discover the project root which helps in discovery of other
+        # publishable items
+        project_root = cmds.workspace(q=True, rootDirectory=True)
+        session_item.properties["project_root"] = project_root
+
+        # if a work file template is defined, add it to the item properties so
+        # that it can be used by attached publish plugins
+        work_template_setting = settings.get("Work file Template")
+        if work_template_setting:
+
+            work_template = publisher.engine.get_template_by_name(
+                work_template_setting.value)
+
+            # store the template on the item for use by publish plugins. we
+            # can't evaluate the fields here because there's no guarantee the
+            # current session path won't change once the item has been created.
+            # the attached publish plugins will need to resolve the fields at
+            # execution time.
+            session_item.properties["work_file_template"] = work_template
+            self.logger.debug("Work file template defined for Maya collection.")
+
+        self.logger.info("Collected current Maya scene")
+
+        return session_item
+
+    def collect_alembic_caches(self, parent_item, project_root):
```

Comment:
maybe this is overthinking - but thinking about the story where say a studio would want to control the location of their 'work alembic' caches using a template. That would just be a matter of subclassing and override this method right? that's pretty neat.